composer/callbacks/generate.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Periodically log generations from a set of prompts."""
+from typing import Any, List, Union, cast
+
+from composer.callbacks.utils import create_interval_scheduler
+from composer.core import Callback, Event, State, Time, get_precision_context
+from composer.loggers import Logger
+from composer.models import HuggingFaceModel
+from composer.utils import dist
+from composer.utils.import_helpers import MissingConditionalImportError
+
+
+class Generate(Callback):
+    """Periodically log generations from a set of prompts.
+
+    Args:
+        prompts (List[str]): The list of prompts you would like to produce generations for
+        interval (Union[str, int, :class:`.Time`]): The interval describing how often checkpoints should be
+            saved. If an integer, it will be assumed to be in :attr:`.TimeUnit.EPOCH`.
+            Otherwise, the unit must be either :attr:`.TimeUnit.EPOCH`, :attr:`.TimeUnit.BATCH`,
+            :attr:`.TimeUnit.TOKEN`, or :attr:`.TimeUnit.SAMPLE`.
+        batch_size (int): Size of a prompt batch for generation.
+        kwargs: All kwargs well be passed along to the call to generate. This is for things like `do_sample`, `top_p`, etc
+    """
+
+    def __init__(self, prompts: List[str], interval: Union[str, int, Time], batch_size: int, **kwargs: Any):
```

Comment:
Should we make batch size optional and default to the number of prompts (i.e. one batch)?

Reply:
good idea!